src/integer/poly_over_z/coefficient_embedding.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright © 2023 Marvin Beckmann
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This modules contains implementations to transform a [`PolyOverZ`]
+//! into a [`MatZ`] and reverse by using the coefficient embedding.
+
+use crate::{
+    integer::{MatZ, PolyOverZ},
+    traits::{
+        FromCoefficientEmbedding, GetCoefficient, GetEntry, GetNumRows, IntoCoefficientEmbedding,
+        SetCoefficient, SetEntry,
+    },
+};
+
+impl IntoCoefficientEmbedding<MatZ> for PolyOverZ {
+    /// Computes the coefficient embedding of the polynomial
+    /// in a [`MatZ`] as a column vector, where the i-th entry
+    /// of the vector corresponds to the i-th coefficient.
+    ///
+    /// Parameters:
+    /// - `size`: determines the length of the vector
+    ///
+    /// Returns a coefficient embedding as a vector
+    ///
+    /// # Examples
+    /// ```
+    /// use std::str::FromStr;
+    /// use qfall_math::{
+    ///     integer::{MatZ, PolyOverZ},
+    ///     traits::IntoCoefficientEmbedding,
+    /// };
+    ///
+    /// let poly = PolyOverZ::from_str("3  17 3 -5").unwrap();
+    /// let vector = poly.into_coefficient_embedding(4);
+    /// let cmp_vector = MatZ::from_str("[[17],[3],[-5],[0]]").unwrap();
+    /// assert_eq!(cmp_vector, vector)
+    /// ```
+    ///
+    /// # Panics ...
+    /// - if `size` is not larger than the degree of the polynomial, i.e.
+    /// the coefficients can not be embedded correctly.
+    fn into_coefficient_embedding(&self, size: impl Into<i64>) -> MatZ {
+        let size = size.into();
+        assert!(size > self.get_degree());
+        let mut out = MatZ::new(size, 1);
+        for j in 0..size {
+            match self.get_coeff(j) {
+                Ok(value) => out.set_entry(j, 0, &value).unwrap(),
```

Comment:
```suggestion
                Ok(value) => out.set_entry(j, 0, value).unwrap(),
```
You don't need ownership of value, therefore you can give it to set_entry. This will make it more efficient if at some point set_entry uses the ownership it gets.